2018---The REAL First Power Up/src/org/usfirst/frc/team2586/robot/Robot.java

```diff
@@ -565,13 +568,43 @@ public void smartDash() {
 		SmartDashboard.putBoolean("lowSwitch", liftLow.get());
 
 		// Gyro
-		SmartDashboard.putNumber("Gyro Angle", gyro.getYaw());
+		SmartDashboard.putNumber("Gyro Angle", gyro.getAngleZ());
 
 		// Get Selected Auton Program
 		SmartDashboard.putString("Auto Program", autoChooser.getSelected());
 
 		// Get Field data from FMS
 		gameData = DriverStation.getInstance().getGameSpecificMessage().toUpperCase();
 	}
+	//function to keep robot tracking straight during teleop while joysticks are in similar positions
+	public void teleopGyro(double leftStick, double rightStick) {
+		double error = 1.0;
```

Comment:
Overall I like the approach here. The tracking state bit is perfect.

This error margin is probably a bit too large. Perhaps .3 or so would make more sense?  Otherwise the driver might not be able to perform a pivot turn and is probably limited to dead spin turns.

Reply:
yeah, John(our driver) worked with me to adjust it and he likes 0.2, 1 was an exaggeration to see that it worked correctly